pkg/util/commands.go

```diff
@@ -269,14 +269,17 @@ func (c *Command) run() (string, error) {
 // PathWithBinary Sets the $PATH variable. Accepts an optional slice of strings containing paths to add to $PATH
 func PathWithBinary(paths ...string) string {
 	path := os.Getenv("PATH")
-	binDir, _ := JXBinLocation()
-	answer := path + string(os.PathListSeparator) + binDir
 	mvnBinDir, _ := MavenBinaryLocation()
+	answer := ""
 	if mvnBinDir != "" {
 		answer += string(os.PathListSeparator) + mvnBinDir
 	}
 	for _, p := range paths {
-		answer += string(os.PathListSeparator) + p
+		if p != "" {
+			answer += string(os.PathListSeparator) + p
```

Comment:
maybe assign this to something new like "customPaths" or "providedPath"? The last line of this function is quite confusing imo. It takes a while to understand what and how you put things together. WDYT?

Reply:
I'll make that a bit clearer to read and update the godoc